pysb/simulator/base.py

```diff
@@ -501,43 +532,52 @@ class SimulationResult(object):
     8.888889   0.000138    4.995633
     13.333333  0.000002    4.999927
     """
-    def __init__(self, simulator, tout, trajectories, squeeze=True,
+    def __init__(self, simulator, tout, trajectories=None,
+                 observables_and_expressions=None,
+                 squeeze=True,
                  simulations_per_param_set=1):
         simulator._logger.debug('SimulationResult constructor started')
         self.squeeze = squeeze
-        self.tout = tout
+        self.tout = np.asarray(tout)
         self._yfull = None
         self._model = simulator._model
         self.n_sims_per_parameter_set = simulations_per_param_set
 
-        # Validate incoming trajectories
-        if hasattr(trajectories, 'ndim') and trajectories.ndim == 3:
-            # trajectories is a 3D array, create a list of 2D arrays
-            # This is just a view and doesn't copy the data
-            self._y = [tr for tr in trajectories]
+        if trajectories is None and observables_and_expressions is None:
+            raise ValueError('Need to supply at least one of species '
+                             'trajectories or observables_and_expressions')
+
+        if trajectories is not None and len(trajectories) > 0:
+            # Validate incoming trajectories
+            if hasattr(trajectories, 'ndim') and trajectories.ndim == 3:
+                # trajectories is a 3D array, create a list of 2D arrays
+                # This is just a view and doesn't copy the data
+                self._y = [tr for tr in trajectories]
+            else:
+                # Not a 3D array, check for a list of 2D arrays
+                try:
+                    if any([tr.ndim != 2 for tr in trajectories]):
```

Comment:
No need for `[ ]` around the comprehension, as `any` accepts a generator expression as well.